graylog2-server/src/main/java/org/graylog2/rest/models/messages/responses/DecorationStats.java

```diff
@@ -53,9 +53,10 @@
     @JsonProperty(FIELD_CHANGED_FIELDS)
     public Map<String, Object> changedFields() {
         return Sets.intersection(originalMessage().keySet(), decoratedMessage().keySet())
-            .stream()
-            .filter(key -> !originalMessage().get(key).equals(decoratedMessage().get(key)))
-            .collect(Collectors.toMap(Function.identity(), key -> originalMessage().get(key)));
+                .stream()
+                .filter(key -> !originalMessage().get(key).equals(decoratedMessage().get(key)))
+                .collect(Collectors.toMap(Function.identity(), key
+                        -> ChangedField.builder().before(originalMessage().get(key)).after(decoratedMessage().get(key)).build()));
```

Comment:
I like the approach to use the new `ChangedField` object to keep track of the original and new values. I think it fits well.

Question: I think this `DecoratorStats` class is used in other parts of Graylog. This like appears to change the map values from the original scalar message field values to instances of the `ChangedField` class. Is this compatible with the other usages of the `DecoratorStats` class? For example, https://github.com/Graylog2/graylog2-server/blob/f35df42e165ac570b8b27de3f8eeac85e74ed610/graylog2-server/src/main/java/org/graylog2/decorators/DecoratorProcessorImpl.java#L116 which is used as part of the [Search Decorators](https://docs.graylog.org/v1/docs/decorators) feature?

Reply:
@danotorrey I was unaware `DecoratorStats` was being used elsewhere. Since it was not being populated at all in normal pipeline simulator diff or the illuminate simulator diff, I just assumed it was safe to make changes to the class. I will check out the features you mentioned as well as confirm there is nowhere else that may be affected with the change. Thank you for pointing this out!